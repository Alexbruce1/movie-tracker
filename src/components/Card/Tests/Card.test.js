import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from '../';

describe('Card', () => {
  describe('Component', () => {
    let wrapper;
    beforeEach(() => {
      const mockMovie = {
        movie: {
          title: 'Movie Title',
          poster: 'Movie Poster',
          overview: 'Movie Overview',
          releaseData: 'Movie Release Date',
          rating: 'Movie Rating'
        }}
      wrapper = shallow(<Card movie={mockMovie}
        changeFavorites={jest.fn()}
        addToFavorites={jest.fn()}
        user={{}}/>);
    });

    it('Should render like snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('Should return the user object from state', () => {
      const mockState = {
        user: { id: 1, loggedIn: true },
        movies: ['movie', 'movie']
      }
      const expected = { user: {id: 1, loggedIn: true }}
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    beforeEach(() => {
      dispatch = jest.fn();
    });
  });
});
