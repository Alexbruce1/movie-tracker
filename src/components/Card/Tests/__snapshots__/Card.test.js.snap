// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movie={
      Object {
        "movie": Object {
          "overview": "Movie Overview",
          "poster": "Movie Poster",
          "rating": "Movie Rating",
          "releaseData": "Movie Release Date",
          "title": "Movie Title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-inner-wrapper"
      >
        <h3
          className="movie-title"
        />
        <p />
        <p>
          Opens: 
        </p>
        <p>
          Viewer Rating: 
        </p>
      </div>,
      "className": "Card",
      "style": Object {
        "backgroundImage": "url(undefined)",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="movie-title"
          />,
          <p />,
          <p>
            Opens: 
          </p>,
          <p>
            Viewer Rating: 
          </p>,
        ],
        "className": "card-inner-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "movie-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Opens: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Opens: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewer Rating: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Viewer Rating: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-inner-wrapper"
        >
          <h3
            className="movie-title"
          />
          <p />
          <p>
            Opens: 
          </p>
          <p>
            Viewer Rating: 
          </p>
        </div>,
        "className": "Card",
        "style": Object {
          "backgroundImage": "url(undefined)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="movie-title"
            />,
            <p />,
            <p>
              Opens: 
            </p>,
            <p>
              Viewer Rating: 
            </p>,
          ],
          "className": "card-inner-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movie-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Opens: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Opens: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewer Rating: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Viewer Rating: ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
